tests/Tests.AzureAppConfiguration/FeatureManagementTests.cs

```diff
@@ -1129,5 +1144,66 @@ Response<ConfigurationSetting> GetTestKey(string key, string label, Cancellation
         {
             return Response.FromValue(TestHelpers.CloneSetting(FirstKeyValue), new Mock<Response>().Object);
         }
+
+        [Fact]
+        public void WithRequirementType()
+        {
+            var emptyFilters = "[]";
+            var nonEmptyFilters = @"[
+                {
+                    ""name"": ""FilterA"",
+                    ""parameters"": {
+                        ""Foo"": ""Bar""
+                    }
+                },
+                {
+                    ""name"": ""FilterB""
+                }
+            ]";
+            var featureFlags = new List<ConfigurationSetting>()
+            {
+                _kv2,
+                featureWithRequirementType("Feature_NoFilters", "All", emptyFilters),
+                featureWithRequirementType("Feature_RequireAll", "All", nonEmptyFilters),
+                featureWithRequirementType("Feature_RequireAny", "Any", nonEmptyFilters)
+            };
+
+            var mockResponse = new Mock<Response>();
+            var mockClient = new Mock<ConfigurationClient>(MockBehavior.Strict);
+
+            mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
+                .Returns(new MockAsyncPageable(featureFlags));
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.ClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object);
+                    options.UseFeatureFlags();
+                })
+                .Build();
+
+            Assert.Null(config["FeatureManagement:MyFeature2:RequirementType"]);
+            Assert.Null(config["FeatureManagement:Feature_NoFilters:RequirementType"]);
```

Comment:
#### TL;DR
I have a question that how should we parse from as below, requirement_type is specified but client_filters is empty:
```ts
conditions: {
  requirement_type: all/any,
  client_filters: []
}
```
Option A) `FeatureManagement:FeatureId = True` or Option B) `FeatureManagement:FeatureId:RequirementType=All/Any`?

#### Thoughts
In previous impl, if `client_filters` is empty, property `FeatureManagement:FeatureId` is set to `True` and skip further processing, because `client_filters` is the only member of `FeatureConditions`.
Two ways to go:
- Now `RequirementType` doesn't make any sense if there's no filter at all. Pick Option A.
- We do not consider any semantic/logic stuffs, simply do parsing work. Pick Option B.

Reply:
On AppConfig side, we look at the `enabled` property of a feature flag to decide whether a feature is enabled, disabled or conditional. When the provider parses a feature flag with `enabled=true` and non-empty `client_filters` list, it means that the feature is conditional, i.e., we need to parse all the filters and `requirement_type`. But if `enabled=true` but `client_filters` list is empty, it means that the feature is always enabled.

However, on feature management side, features without any filters are considered disabled. So when a feature is supposed to be enabled (i.e., `FeatureManagement:FeatureId = True`), we automatically inject a special filter called "AlwaysOn" to enable the feature ([code](https://github.com/microsoft/FeatureManagement-Dotnet/blob/5c3e941afc0d1e88a57253364fbce61b2138394f/src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs#L144-L156)).

So from your example, we cannot go for Option B without making changes in the Feature Management library to handle this scenario. Option A should be fine to implement here.
Thoughts @jimmyca15 @rossgrambo ?

Reply:
Option a is my vote

Reply:
It makes sense to me now. This PR already follows Option A, and I'll keep it as is.

Thank you all for elaborated explanations.